### ISSUE TITLE
pythonnet or not?

### ISSUE THREAD

**Issue by astronouth7303** [r0x0r/pywebview#60]
Since no matter what, you need pywin32 and comtypes, is there an appreciable difference to users between using pythonnet (and the .NET API) or not (and the Win32/COM API)?

**Comment by denfromufa** [comment]
pythonnet can also target WPF

**Comment by astronouth7303** [comment]
Does that make a difference, given that all it does is load MSHTML?

**Comment by r0x0r** [comment]
Pythonnet does not require pywin32 or comtypes. However, if you freeze it both pywin32 and pythonnet libraries will be bundled provided you have them installed. 
There is no difference from a developer point of view, which implementation you use, unless you are gonna target a really old Windows version that does not have .net included. Both use MSHTML in the end. The original motivation for the net implementation was Edge support, but as it turned out Edge is supported only in UWP apps. One can hope that Microsoft will backport Edge support to winforms. Comtypes compatibility problems with Python 3.5 until recently, so getting rid of this dependency was another factor for an alternative implementation 
The net version  is quite recent and is not tested properly. The application icon is missing or freezing with pyinstaller does not work out of the box, for instance. However, given how much easier is development is with .net comparing to win32, I am going to focus on the former. For example currently I have no plans for implementing load_html function for win32
The net implementation uses Windows Forms, not WPF by the way. I see no point in WPF support. UWP would be nice, but it is not possible at the moment. 

To summarize all this, you can target either one, but the development of pywebview will be focused around .net in the future. I would love to hear about your experience with the net implementation.

**Comment by astronouth7303** [comment]
Oh, then there's another bug: `winforms.py` imports a few things from `win32.py` and (indirectly) makes use of both `pywin32` and `comtypes`. This does not look fixed on master.

**Comment by r0x0r** [comment]
True, winforms.py imports a few things from win32.py and win_gen.py, but
they are no dependant on pywin32 or comtypes. What indirect use do you mean?

On 29 Oct 2016 12:53 a.m., "Jamie Bliss" <notifications@github.com> wrote:

Oh, then there's another bug: winforms.py imports a few things from win32.py
and (indirectly) makes use of both pywin32 and comtypes. This does not look
fixed on master.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/r0x0r/pywebview/issues/60#issuecomment-257038132>, or mute
the thread

**Comment by r0x0r** [comment]
My bad, I see the bug now. I will refactor the code and get rid of unnecessary dependencies.

**Comment by r0x0r** [comment]
Fixed in master. Winforms should not have any pywin32 or comtypes dependencies now.

**Comment by r0x0r** [comment]
Fixed in 1.3
